File: ios/MyDeviceInfo.m
#import <Foundation/Foundation.h>
#import "React/RCTBridgeModule.h"
#import "React/RCTEventEmitter.h"

@interface RCT_EXTERN_MODULE(MyDeviceInfo, NSObject)

RCT_EXTERN_METHOD(getDeviceInfo:(NSString *)message
                  resolve:(RCTPromiseResolveBlock)resolve
                  reject:(RCTPromiseRejectBlock)reject)

@end

File: ios/MyDeviceInfo.swift
import Foundation
import React

@objc(MyDeviceInfo)
class MyDeviceInfo: NSObject {

  @objc func getDeviceInfo(_ message: String, resolve: @escaping RCTPromiseResolveBlock, reject: @escaping RCTPromiseRejectBlock) {
    let response = "Received \(message) from native module"
    resolve(["response": response])
  }
  
}
